apps/gitness/src/pages-v2/repo/repo-sidebar.tsx

```diff
@@ -102,38 +107,48 @@ export const RepoSidebar = () => {
         }
       }).then(response => {
         if (response.body.type === 'dir') {
-          navigate(`/spaces/${spaceId}/repos/${repoId}/code/new/${gitRef || selectedBranch.name}/~/${fullResourcePath}`)
+          navigate(`/${spaceId}/repos/${repoId}/code/new/${gitRef || selectedBranch.name}/~/${fullResourcePath}`)
         } else {
           const parentDirPath = fullResourcePath?.split(FILE_SEPERATOR).slice(0, -1).join(FILE_SEPERATOR)
-          navigate(`/spaces/${spaceId}/repos/${repoId}/code/new/${gitRef || selectedBranch.name}/~/${parentDirPath}`)
+          navigate(`/${spaceId}/repos/${repoId}/code/new/${gitRef || selectedBranch.name}/~/${parentDirPath}`)
         }
       })
     } else {
-      navigate(`/spaces/${spaceId}/repos/${repoId}/code/new/${gitRef || selectedBranch.name}/~/`)
+      navigate(`/${spaceId}/repos/${repoId}/code/new/${gitRef || selectedBranch.name}/~/`)
     }
   }, [fullResourcePath, gitRef, navigate, repoId, repoRef, selectedBranch.name, spaceId])
 
   const navigateToFile = useCallback(
     (filePath: string) => {
-      navigate(`/spaces/${spaceId}/repos/${repoId}/code/${gitRef || selectedBranch.name}/~/${filePath}`)
+      navigate(`/${spaceId}/repos/${repoId}/code/${gitRef || selectedBranch.name}/~/${filePath}`)
     },
     [gitRef, selectedBranch.name, navigate, repoId, spaceId]
   )
 
+  // TODO: repoId and spaceId must be defined
+  if (!repoId || !spaceId) return <></>
+
   return (
     <>
       <RepoSidebarView
         hasHeader
         hasSubHeader
+        repoId={repoId}
+        spaceId={spaceId}
         selectedBranch={selectedBranch}
         selectBranch={selectBranch}
         branchList={branchList}
         tagList={[]}
+        // TODO: new props navigateToNewFolder
+        navigateToNewFolder={() => {}}
         navigateToNewFile={navigateToNewFile}
         navigateToFile={navigateToFile}
         filesList={filesList}
-        // repoDetails={repoDetails?.body}
-      />
+      >
+        {repoDetails?.body?.content?.entries?.length && (
```

Comment:
In the event that `repoDetails.body.content.length` is `0`, this will output `0` in the HTML; worth casting to a boolean to make sure it won't
```suggestion
        {!!repoDetails?.body?.content?.entries?.length && (
```